saltproc/serpent_depcode.py

```diff
@@ -175,20 +188,33 @@ def convert_nuclide_code_to_name(self, nuc_code):
         """
 
         if '.' in str(nuc_code):
-            nuc_code = pyname.zzzaaa_to_id(int(nuc_code.split('.')[0]))
+            nuc_code = int(nuc_code.split('.')[0])
+            if self.zaid_convention == 'serpent':
+                nuc_code = pyname.zzzaaa_to_id(nuc_code)
+            if self.zaid_convention in ('mcnp', 'nndc'):
+                if self.zaid_convention == 'mcnp' and nuc_code in (95242, 95642):
+                    if nuc_code == 95242:
+                        nuc_code = 95642
+                    else:
+                        nuc_code = 95242
```

Comment:
This has nothing to do with OpenMC. It is all about where the nuclide cross section data is sourced from, since their ZAIDs will have slightly different conventions for metastable isotopes. This is a bit confusing so I'm going to go through this step-by-step.

1. Recall that the standard `nndc` convention is identical to the `mcnp` convention, *except* in the case of Am-242 and Am-242m. The `mcnp` convention for ZAIDs of metastable isotopes is ZA = Z * 1000 + (A + 300 + 100*m). This means that Ag-110m has ZA = 47510, and Am-242m has ZA = 95642. In the `nndc` convention, for whatever reason, the ZA for Am-242m and Am242 are swapped, so ZA(Am242) = 95642, and ZA(Am242m) = 95242.
2. PyNE's `nucname` module contains functions for handling ZAID codes. One such function is `nucname.mcnp_to_id`, which converts the ZAID code in `mcnp` convention to PyNEs internal id format. However, during testing, I noticed that this function actually converts ZAID codes in `nndc`  convention! That is, will convert `95242` to `952420001`, and `95642` to `952420000`, but convert everything else normally. So if we are using the `mcnp` convention, we'll need to input the wrong ZAID code to get the right PyNE id. Does that make sense?

Reply:
That all mostly makes sense (I meant to say MCNP, but had OpenMC on my mind when I made the comment, apologies). Just to make sure I understand this correctly:

In MCNP: 
- Am242 = 95242
- Am242m = 95642
In NNDC:
- Am242 = 95642
- Am242m = 95242.

Your function here says that if you are using MCNP and you give it 95242 (Am242 in MCNP), then it converts it to 95642 (Am242m in MCNP). If you give it 95642 (Am242m in MCNP), then it converts it to 95242 (Am242). So that gets switched. This assumes that you are providing NNDC nomenclature and trying to use MCNP. What if you are using MCNP nomenclature already and don't need to switch it?

Reply:
You have the first right. 
> This assumes that you are providing NNDC nomenclature and trying to use MCNP. What if you are using MCNP nomenclature already and don't need to switch it?

The reason we do the swap is because PyNE assumes an `nndc` format for `nucname.mcnp_to_id` and other mcnp-related functions. That is, there is a discrepancy between our understanding of the `mncp` convention and what PyNE's understanding of the `mcnp` convention. PyNE's `mcnp` convention is equivalent to *our* `nndc` convention. So if we want to use *our* `mcnp` convention, we have to swap the ZAIDs for Am242 and Am242m that we input into the mcnp-related functions in PyNE to get the correct outputs (ZAIDs and nuclide names).

Reply:
That makes sense. I'm making sure that there is no situation in which these ZAIDs would be switched when they don't have to be.